src/webgpu/api/operation/sampling/anisotropy.spec.ts

```diff
@@ -6,14 +6,310 @@ Note that anisotropic filtering is never guaranteed to occur, but we might be ab
+
+    const bufferSize = kBytesPerRow * textureSize; // RGBA8 for each pixel (256 > 16 * 4)
+
+    // init checkerboard texture data
+    const data: Uint8Array = new Uint8Array(bufferSize);
+    for (let r = 0; r < textureSize; r++) {
+      const o = r * kBytesPerRow;
+      for (let c = o, end = o + textureSize * 4; c < end; c += 4) {
+        const cid = (r + (c - o) / 4) % 2;
+        const color = checkerColors[cid];
+        data[c] = color[0];
+        data[c + 1] = color[1];
+        data[c + 2] = color[2];
+        data[c + 3] = color[3];
+      }
+    }
+    const buffer = t.makeBufferWithContents(
+      data,
+      GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST
+    );
+    const bytesPerRow = kBytesPerRow;
+    const rowsPerImage = textureSize;
+
+    textureEncoder.copyBufferToTexture(
+      {
+        buffer,
+        bytesPerRow,
+        rowsPerImage,
+      },
+      {
+        texture,
+        mipLevel: 0,
+        origin: [0, 0, 0],
+      },
+      [textureSize, textureSize, 1]
+    );
+
+    t.device.defaultQueue.submit([textureEncoder.finish()]);
+
+    const textureView = texture.createView();
+    const byteLength = kRTSize * kBytesPerRow;
+    const results: Uint8Array[] = [];
+
+    for (const maxAnisotropy of [1, 16, 1024]) {
+      const sampler = t.device.createSampler({
+        magFilter: 'linear',
+        minFilter: 'linear',
+        mipmapFilter: 'linear',
+        maxAnisotropy,
+      });
+      const d = t.createAlignedCopyForMapRead(
+        t.copyRenderTargetToBuffer(t.drawSlantedPlane(textureView, sampler)),
+        byteLength,
+        0
+      ).dst;
+      await d.mapAsync(GPUMapMode.READ);
+      results.push(new Uint8Array(d.getMappedRange()));
+    }
+
+    const check0 = t.checkBuffer(results[0], results[1]);
+    if (check0 === undefined) {
+      t.warn('Render results with sampler.maxAnisotropy being 1 and 16 should be different.');
+    }
+    const check1 = t.checkBuffer(results[1], results[2]);
+    if (check1 !== undefined) {
+      t.expect(
+        false,
+        'Render results with sampler.maxAnisotropy being 16 and 1024 should be the same.'
+      );
+    }
+  });
+
+g.test('anisotropic_filter_mipmap_color')
+  .desc(
+    `Anisotropic filter rendering tests that draws a slanted plane and samples from a texture
+    containing mipmaps of different colors. Given the same fragment with dFdx and dFdy for uv being different,
+    sampler with bigger maxAnisotropy value tends to bigger mip levels to provide better details.
+    We can then look at the color of the fragment to know which mip level is being sampled from and to see
+    if it fits expectations.
+    A similar webgl demo is at https://jsfiddle.net/t8k7c95o/5/`
+  )
+  .params([
+    {
+      maxAnisotropy: 1,
+      _results: [
+        { coord: { x: xMiddle, y: 2 }, expected: colors[2] },
+        { coord: { x: xMiddle, y: 6 }, expected: [colors[0], colors[1]] },
+      ],
+      generateWarningOnly: false,
```

Comment:
nit: Should be a private param (or maybe it should even be per-entry in the _results array)